server/schemas/schemaModel.js

```diff
@@ -28,32 +28,26 @@ module.exports = {
         //console.debug("findOneAsync args:", arguments);
         if (schema) {
           error = {message: "Schema already exists"};
-          helpers.errorHandler(error, request, response, next);
-          return;
+          next(error, false);
         } else {
-          Model.createAsync({
+          return Model.createAsync({
```

Comment:
Brian is on the money here. Mongoose has a Model.create function, and bluebird's PromisifyAll function extends a promise version of the the method with "async" appended to the method name.

Mongoose Model.create: http://mongoosejs.com/docs/models.html
Bluebird PromisifyAll: https://github.com/petkaantonov/bluebird/blob/master/API.md#promisepromisifyallobject-target--object-options---object